src/mpi/coll/reduce/reduce.c

```diff
@@ -321,6 +334,94 @@ int MPIR_Reduce_impl(const void *sendbuf, void *recvbuf, int count,
     goto fn_exit;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIR_Reduce_decision_tree
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+int MPIR_Reduce_decision_tree(const void *sendbuf, void *recvbuf, int count, MPI_Datatype datatype,
+                              MPI_Op op, int root, MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag)
+{
+    int mpi_errno = MPI_SUCCESS;
+    size_t size;
+    cJSON *device_info;
+
+    /* TODO - Calculate size of buffer */
+
+    mpi_errno = MPIU_Decision_tree_traverse("reduce", comm_ptr, size, datatype, coll_info);
```

Comment:
What does this function return? `coll_info` is not used elsewhere.

Reply:
Sorry, I changed the name of that variable. `s/device_info/coll_info/`.